jupyter/analysis-automation/BasicExample.ipynb

```diff
@@ -0,0 +1,271 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import asyncio\n",
+    "from systemlink.clients.nitdmreader.api.metadata_api import MetadataApi\n",
+    "from systemlink.clients.nitdmreader.models.file_list import FileList\n",
+    "\n",
+    "metadata_api = MetadataApi()\n",
+    "import scrapbook as sb\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "from systemlink.clients.nitdmreader import *\n",
+    "data_api = DataApi()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# --- Metadata helper methods ---\n",
+    "\n",
+    "def get_groups(file_info):\n",
+    "    if not hasattr(file_info, '__getitem__'):\n",
+    "        return [g for g in file_info.groups]\n",
+    "    else:\n",
+    "        return [g for g in file_info[\"groups\"]]\n",
+    "\n",
+    "def get_group_names(file_info):\n",
+    "    if not hasattr(file_info, '__getitem__'):\n",
+    "        return [g.name for g in get_groups(file_info)]\n",
+    "    else:\n",
+    "        return [g[\"name\"] for g in get_groups(file_info)]\n",
+    "\n",
+    "def get_channels(group):\n",
+    "    if not hasattr(group, '__getitem__'):\n",
+    "        return [c for c in group.channels]\n",
+    "    else:\n",
+    "        return [c for c in group[\"channels\"]]\n",
+    "\n",
+    "def get_channel_names(group):\n",
+    "    if not hasattr(group, '__getitem__'):\n",
+    "        return [c.name for c in get_channels(group)]\n",
+    "    else:\n",
+    "        return [c[\"name\"] for c in get_channels(group)]\n",
+    "\n",
+    "def get_property(element, property_name):\n",
+    "    if not hasattr(element, '__getitem__'):\n",
+    "        return next((e.value for e in element.properties.properties if e.name == property_name), None)\n",
+    "    else:\n",
+    "        return next((e[\"value\"] for e in element[\"properties\"][\"properties\"] if e[\"name\"] == property_name), None)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# use data links to read data\n",
+    "data_links = ni_analysis_automation[\"data_links\"]\n",
+    "file_ids = [d[\"fileId\"] for d in data_links]\n",
+    "file_infos = await metadata_api.get_multiple_file_info(FileList(file_ids))\n",
+    "file_info = file_infos[0]\n",
+    "\n",
+    "test_file_name = get_property(file_info, \"name\")\n",
+    "program_name = get_property(file_info, \"Test~Procedure\") or get_property(file_info, file_name_prop)\n",
+    "#display(program_name)\n",
+    "group_names = get_group_names(file_info)\n",
+    "#display(group_names)\n",
+    "\n",
+    "stati = []\n",
+    "channels = []\n",
+    "formatted_properties = []\n",
+    "\n",
+    "grp_idx = 0\n",
+    "for group in get_groups(file_info):\n",
+    "    max_values_of_group = []    \n",
```

Comment:
There are some additional blanks in this code. Maybe we should format the Python code in this notebook with something like [the JupyterLab Code Formatter](https://jupyterlab-code-formatter.readthedocs.io/en/latest/index.html) and [black](https://jupyterlab-code-formatter.readthedocs.io/en/latest/installation.html#installation-step-2-installing-a-supported-code-formatter) as the formatter.

Reply:
Have you tried to auto-format the notebook with black? Looks like there are still some additional blanks.